tests/on_target/tests/test_power.py

```diff
@@ -0,0 +1,229 @@
+##########################################################################################
+# Copyright (c) 2024 Nordic Semiconductor
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+##########################################################################################
+
+import os
+import time
+import json
+import types
+import pytest
+import csv
+import pandas as pd
+import plotly.express as px
+from tests.conftest import get_uarts
+from ppk2_api.ppk2_api import PPK2_API
+from utils.uart import Uart
+from utils.flash_tools import flash_device, reset_device, recover_device
+import sys
+sys.path.append(os.getcwd())
+from utils.logger import get_logger
+
+logger = get_logger()
+
+UART_TIMEOUT = 60 * 30
+POWER_TIMEOUT = 60 * 20
+MAX_CURRENT_PSM_UA = 10
+SAMPLING_INTERVAL = 0.01
+CSV_FILE = "power_measurements.csv"
+HMTL_PLOT_FILE = "power_measurements_plot.html"
+
+SEGGER = os.getenv('SEGGER')
+UART_ID = os.getenv('UART_ID', SEGGER)
+
+def get_uarts():
+    base_path = "/dev/serial/by-id"
+    try:
+        serial_paths = [os.path.join(base_path, entry) for entry in os.listdir(base_path)]
+    except (FileNotFoundError, PermissionError) as e:
+        logger.error(e)
+        return False
+    uarts = []
+    for path in sorted(serial_paths):
+        if UART_ID in path:
+            uarts.append(path)
+        else:
+            continue
+    return uarts
+
+def save_badge_data(average):
+    badge_filename = "power_badge.json"
+    logger.info(f"Minimum average current measured: {average}uA")
+    if average < 0:
+        pytest.fail(f"current cant be negative, current average: {average}")
+    if average <= 10:
```

Comment:
for consistency either make this to elif or the following to if. Or just a blank line in between between the previous if statement and this one.